ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Model.hs

```diff
@@ -374,3 +376,25 @@ between (SecurityParam k) from to m = do
         -> return frag'
         | otherwise
         -> Left $ MissingBlock p
+
+-- Would the garbage collector (if run) be able to garbage collect the block
+-- corresponding to the given point?
+garbageCollectable :: forall blk. HasHeader blk
+                   => SecurityParam -> Model blk -> Point blk -> Bool
+garbageCollectable (SecurityParam k) m@Model{..} p =
+    not onCurrentChain && olderThanK
+  where
+    onCurrentChain = Chain.pointOnChain p (currentChain m)
+    olderThanK     = Chain.pointSlot    p < oldestAllowedSlot
+
+    oldestAllowedSlot = SlotNo $ slotAtTip - min k slotAtTip
```

Comment:
This doesn't look right. This is taking k slots, not k blocks. Immutability / no-rollback is about K blocks, not slots.

What test would have caught this? e.g. in `garbageCollect` below we could have deleted forks that were only k/2 old. Perhaps we want a test that covers switching to a fork at the maximum possible point back, interleaved with GCing, and checking we can still switch since the blocks are still there. Just a thought. You can probably think of a better one, and note it down for later in one of the tickets.

Reply:
Of course, silly me. Thanks for catching this. See #693.